factory-reset: cleanup

antonlacon opened the pull request:
This is a cleanup of the system reset script. Two functional changes:

The hard reset is now filesystem agnostic, meaning it will use rm to clear /storage, instead of recreating the filesystem.

The soft reset now specifically deletes .cache, .config and .kodi only. These are the same directories preserved by LE backups

chewitt commented on packages/mediacenter/LibreELEC-settings/scripts/factory-reset:
```diff
@@ -38,42 +32,37 @@ get_target() {
     StartProgress countdown "Rebooting in 5s... " 5 "NOW"
   fi
-elif [ -f /storage/.cache/reset_xbmc ] ; then
+elif [ -f /storage/.cache/reset_xbmc ]; then
```
As much as I like a trip down memory lane, perhaps we should update/deobfuscate the trigger-file names and use reset_soft/reset_hard instead of reset_oe/reset_xbmc ?

antonlacon replied:
Commit added to do this. I'll PR the necessary change to settings addon shortly.

chewitt commented on packages/mediacenter/LibreELEC-settings/scripts/factory-reset:
```diff
@@ -38,42 +32,37 @@ get_target() {
-      if [ ! -z $uuid ] ; then
-        tune2fs -U $uuid $target &>/dev/null
-      fi
+      rm -rf /storage/* /storage/.* &>/dev/null
```
IIRC the reason for the current approach is time; it takes a couple of seconds to nuke/pave the partition vs. much longer to recursively delete 3-4GB of files from a typical SD card or USB stick.

HiassofT replied:
I'd also prefer to keep the current mkfs approach, it's faster and safer - and also helps to get to a sane state eg after filesystem corruption.

It might be worth though to check if /storage actually is a block device and fall back to rm otherwise - that should allow hard reset to also work when netbooting with /storage on NFS.

BTW: using `/storage/.*` is dangerous, that'll match `.` and `..` as well (rm might prevent you from shooting yourself in the foot, but better don't rely on that).

antonlacon commented:
Added /storage/.ssh to being removed on soft reset as well. This is also included in the backup (it's filtered out when empty and my backup tarball didn't include it).

Changed deleting of files to be handled by `find ... -delete` to avoid the rm issue mentioned above.

Hard reset reformats when it's a block device and has an ext4 filesystem, and fallsback to `find` to delete everything. I don't know if get_target() needs to be adjusted to accommodate netbooting, but I think it's ok. get_target() should return `NFS=/some/thing`, which won't be a block device and `find` will take clear of deleting from there, but I'm not netbooting to confirm this.

Moved the script to use double brackets for tests. I got bit by [ ! -z ... ] not being the same as [ -n ... ], while using [[ ]] is.

HiassofT commented:
Please leave .ssh intact on a soft reset - wiping host keys and authorized_keys results in a mess for everyone using ssh.

mglae commented:
Please keep `[ ... ]`. `[[ .. ]]`is only an incomplete bash extension in ash.

antonlacon commented:
Updated.

antonlacon commented:
Matching PR with settings-addon changes due to reset file name change is https://github.com/LibreELEC/service.libreelec.settings/pull/179

I believe this is otherwise complete.

antonlacon commented:
Rebased.
